perfkitbenchmarker/providers/aws/aws_vpn_network.py

```diff
@@ -0,0 +1,573 @@
+      tunnel_config.endpoints[self.name]['tunnel_id'] = self.vpn_connection.id
+
+    if not tunnel_config.endpoints[target_endpoint]['cidr']:
+      logging.info(
+          "tunnel_config: Target CIDR needed for route creation... returning"
+      )
+      return
+
+    if not tunnel_config.endpoints[self.name]['is_configured']:
+      aws_network.AwsFirewall.GetFirewall().AllowIcmp(None,
+        region=self.region,
+        security_group=net.vpc.default_security_group_id,
+        cidr=tunnel_config.endpoints[target_endpoint]['cidr']
+      )
+      self.vpn_connection.Create_VPN_Cnxn_Route(
+          tunnel_config.endpoints[target_endpoint]['cidr']
+      )
+
+    tunnel_config.endpoints[self.name]['is_configured'] = True
+
+  def IsTunnelReady(self, tunnel_id):
+    return True
+
+  def Create(self):
+    """Creates the actual VPNGW."""
+    benchmark_spec = context.GetThreadBenchmarkSpec()
+    if benchmark_spec is None:
+      raise errors.Error(
+          'GetNetwork called in a thread without a '
+          'BenchmarkSpec.'
+      )
+    # with self._lock:
+    if self.created:
+      return
+    if self.vpn_gateway_resource:
+      self.vpn_gateway_resource.Create()
+    key = self.name
+    # with benchmark_spec.vpn_gateways_lock:
+    if key not in benchmark_spec.vpn_gateways:
+      benchmark_spec.vpn_gateways[key] = self
+    return benchmark_spec.vpn_gateways[key]
+    self.created = True
+
+  def Delete(self):
+    """Deletes the actual VPNGW."""
+    if self.vpn_connection and self.vpn_connection._Exists():
+      self.vpn_connection.Delete()
+
+    if self.customer_gw and self.customer_gw._Exists():
+      self.customer_gw.Delete()
+
+    # vpn_gateways need deleted last
+    if self.vpn_gateway_resource:
+      self.vpn_gateway_resource.Detach()
+      self.vpn_gateway_resource.Delete()
+
+
+class AwsVPNConnection(resource.BaseResource):
+  """An object representing a Aws VPNConnection."""
+
+  def __init__(
+      self,
+      cgw_id,
+      vpg_id,
+      region,
+      psk=None,
+      routing="static",
+      type="ipsec.1"
+  ):
+    super(AwsVPNConnection, self).__init__()
+    self.cgw_id = cgw_id
+    self.vpg_id = vpg_id
+    self.region = region
+    self.type = type
+    self.id = None
+    self.routing = routing
+    self.psk = FLAGS.run_uri
+    self.cnxn_details = None
+    self.ip = None  # do we need to support failover (2nd IP?)
+
+  def _Create(self):
+    """creates VPN cnxn and 2 vpn tunnels to CGW."""
+    create_cmd = util.AWS_PREFIX + [
+        'ec2',
+        'create-vpn-connection',
+        # '--debug',
+        '--region=%s' % self.region,
+        '--customer-gateway-id=%s' % self.cgw_id,
+        '--vpn-gateway-id=%s' % self.vpg_id,
+        '--type=%s' % self.type,
+        '--options=%s' % self._getCnxnOpts()
+    ]
+    response, _ = util.IssueRetryableCommand(create_cmd)
+    response_xml = response[response.find("<"):response.rfind(">") +
```

Comment:
needs more comments about formats and parsing.  Can you parse as Json and then xml rather than this manual stuff?